core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java

```diff
@@ -56,7 +57,7 @@ private FieldMaskingSpanQueryBuilder() {
      */
     public FieldMaskingSpanQueryBuilder(SpanQueryBuilder queryBuilder, String fieldName) {
         this.queryBuilder = Objects.requireNonNull(queryBuilder);
-        this.fieldName = fieldName;
+        this.fieldName = Objects.requireNonNull(fieldName);
```

Comment:
The parser checks for set fieldname, throws QueryParsingException otherwise, so we can already require it here I think.